packages/core/src/toPromise.ts

```diff
@@ -0,0 +1,47 @@
+import { AnyActorRef, OutputFrom } from './types.ts';
+
+/**
+ * Returns a promise that resolves to the `output` of the actor when it is done.
+ *
+ * @example
+ * ```ts
+ * const machine = createMachine({
+ *   // ...
+ *   output: {
+ *     count: 42
+ *   }
+ * });
+ *
+ * const actor = createActor(machine);
+ *
+ * actor.start();
+ *
+ * const output = await toPromise(actor);
+ *
+ * console.log(output);
+ * // logs { count: 42 }
+ * ```
+ */
+export function toPromise<T extends AnyActorRef>(
+  actor: T
+): Promise<OutputFrom<T>> {
+  // TODO: this is typed as `any`
+  const currentSnapshot = actor.getSnapshot();
+
+  if (currentSnapshot.status === 'done') {
+    return Promise.resolve(currentSnapshot.output);
+  }
```

Comment:
This is redundant - we already resolve `complete` callbacks immediately if the actor is already done.

Reply:
I believe that we'll be able to remove this once this lands: https://github.com/statelyai/xstate/pull/4570